lib/galaxy/model/__init__.py

```diff
@@ -204,7 +206,21 @@
 
 _datatypes_registry = None
 
-mapper_registry = registry()
+OPTIONAL_STR_TO_STR_DICT = Optional[Dict[str, str]]
```

Comment:
My strong suggestion is to not include `Optional` here. In the model definition, `Optional` will be used to infer nullability of the database field:
`foo: Mapped[mytype]` and `foo: Mapped[Optional[mytype]]` will generate different database schemas: the former will define foo as `NOT NULL`. Furthermore, this can be overwritten with the `nullable` attribute in `mapped_column`: `foo: Mapped[mytype] = mapped_column(nullable=True)` will generate a definition for a nullable database field (while allowing `None` in the Python app). And vise-versa.
So, seeing or not seeing `Optional` in the model definition is really important.

Reply:
I think it respects the runtime type description and isn't like looking for Optional using raw Python source code reading or something naive like that. 

``create_test.py`` -->

```python
from sqlalchemy.orm import DeclarativeBase
from sqlalchemy.orm import Mapped, mapped_column
from sqlalchemy.schema import CreateTable
from typing import Optional

OptionalFoo = Optional[str]
RequiredFoo = str


class Base(DeclarativeBase):
    pass


class SomeClass(Base):
    __tablename__ = "some_table"

    id: Mapped[int] = mapped_column(primary_key=True)
    name: Mapped[str]
    optional_foo: Mapped[OptionalFoo]
    required_foo: Mapped[RequiredFoo]


print(CreateTable(SomeClass.__table__))
```


```
> python create_test.py 

CREATE TABLE some_table (
	id INTEGER NOT NULL, 
	name VARCHAR NOT NULL, 
	optional_foo VARCHAR, 
	required_foo VARCHAR NOT NULL, 
	PRIMARY KEY (id)
)
```

Reply:
Oh, absolutely. I didn't mean it wouldn't work - I was only referring to keeping the definition explicit and straightforward. Given our 10K+ lines model definition module, the line defining our custom `OptionalFoo` type can be thousands of lines removed from where it's used. The use of "optional" as a prefix for such a type name would be our own naming convention that we are bound to eventually disregard. I think, having Python's `typing.Optional` in the model's definition' makes it more explicit. Given that this one term determines what goes into the database schema, I think it would be better to keep it where the field is defined - i.e., make it less error prone and make the model easier to understand.

Reply:
I don't agree really but I've made the change.

Reply:
Thank you, I do appreciate it! We can revisit this after we have all the type/model improvements in place.